tabular/src/autogluon/tabular/trainer/abstract_trainer.py

```diff
@@ -1263,29 +1297,53 @@ def _train_multi_initial(self, X, y, models: List[AbstractModel], k_fold, n_repe
             time_split = True
         else:
             time_split = False
+        k_fold_start = 0
         if k_fold == 0:
             time_ratio = hpo_time_ratio if hpo_enabled else 1
-            models = self._train_multi_fold(models=models, hyperparameter_tune_kwargs=hyperparameter_tune_kwargs, feature_prune=feature_prune, time_limit=time_limit, time_split=time_split, time_ratio=time_ratio, **fit_args)
+            models = self._train_multi_fold(models=models, hyperparameter_tune_kwargs=hyperparameter_tune_kwargs,
+                                            time_limit=time_limit, time_split=time_split, time_ratio=time_ratio, **fit_args)
         else:
-            k_fold_start = 0
-            if hpo_enabled or feature_prune:
-                time_start = time.time()
+            bagged_time_start = time.time()
+            if hpo_enabled:
                 time_ratio = (1 / k_fold) * hpo_time_ratio
-                models = self._train_multi_fold(models=models, hyperparameter_tune_kwargs=hyperparameter_tune_kwargs, feature_prune=feature_prune,
-                                                k_fold_start=0, k_fold_end=1, n_repeats=n_repeats, n_repeat_start=0, time_limit=time_limit, time_split=time_split, time_ratio=time_ratio, **fit_args)
+                models = self._train_multi_fold(models=models, hyperparameter_tune_kwargs=hyperparameter_tune_kwargs,
+                                                k_fold_start=0, k_fold_end=1, n_repeats=n_repeats, n_repeat_start=0, time_limit=time_limit,
+                                                time_split=time_split, time_ratio=time_ratio, **fit_args)
                 k_fold_start = 1
-                if time_limit is not None:
-                    time_limit = time_limit - (time.time() - time_start)
+            bagged_time_limit = time_limit - (time.time() - bagged_time_start) if time_limit is not None else None
+            models = self._train_multi_fold(models=models, hyperparameter_tune_kwargs=None, k_fold_start=k_fold_start,
+                                            k_fold_end=k_fold, n_repeats=n_repeats, n_repeat_start=0, time_limit=bagged_time_limit, **fit_args)
 
-            models = self._train_multi_fold(models=models, hyperparameter_tune_kwargs=None, feature_prune=False, k_fold_start=k_fold_start, k_fold_end=k_fold, n_repeats=n_repeats, n_repeat_start=0, time_limit=time_limit, **fit_args)
+        multi_fold_time_elapsed = time.time() - multi_fold_time_start
+        if time_limit is not None:
+            time_limit = time_limit - multi_fold_time_elapsed
 
+        if feature_prune and len(models) > 0:
+            feature_prune_time_start = time.time()
+            candidate_features = self._proxy_model_feature_prune(feature_prune_kwargs, time_limit, multi_fold_time_elapsed, kwargs['level'], X.columns.tolist())
```

Comment:
In case if `multimodal` setup is active, please ensure that the columns marked for other modalities are not removed (i.e. images names might be useless feature, but the image itself is a different story).